models/articles.model.js

```diff
@@ -13,4 +13,20 @@ exports.readArticle = (article) => {
         else
         return{"finishedArticle" : currentArticle}
     })
+}
+
+exports.fetchArticles = () => {
+    return db.query
+    (` ALTER TABLE articles ADD COLUMN comments_count INT DEFAULT 0 NOT NULL;`)
+    .then(() => {
+        return db.query
+   (`UPDATE articles SET comments_count = (SELECT COUNT(*) FROM comments WHERE comments.article_id = articles.article_id);`)
+    })
+    .then(() => {
+        return db.query
+   (`SELECT author,title,article_id,topic,created_at,votes,article_img_url,comments_count FROM articles ORDER BY created_at DESC;`)
+    })
+    .then(({rows}) => {
+    return{"allArticles": rows}
+   })
```

Comment:
This unfortunately is not quite what the ticket is asking for. Here you are directly altering the database - what we would like you to do is essentially do all of this inside a select. So the comment_count column is generated through a query and not added to the database. 

You have the code here basically - just see if you can get it working within the select on line 27 and without the alter table/update!

Reply:
Does it need a join? Struggling to make it work!